javascript/es6-functions.md

```diff
@@ -0,0 +1,202 @@
+## ECMAScript 6 Functions
+
+In this TIL, we will look at three features in ES6 regarding functions:
+* Default Values
+* Block level functions
+* Arrow functions
+
+#### Default Values
+
+JavaScript functions allow any number of parameters to be passed regardless of
+the number of declared parameters. To have certain values in the parameters
+represent defaults, you would need to do this:
+
+```javascript
+function doRequest(url, timeout, callback) {
+  var timeout = timeout || 2000;
+  var callback = callback || function() {
+    // callback code
+  };
+
+  // rest of code
+}
+```
+If `timeout` is not passed as an argument, then it would be set to 2000. Likewise
+for `callback`. With ES6, we can now set default arguments.
+
+```javascript
+function doRequest(url, timeout = 2000, callback) {
+  // rest of code
+}
+```
+
+The default value for timeout will only be used when there is no second argument
+present. For the last argument, we can also define a default function for it.
+
+```javascript
+function someCallback() {
+  return function() {
+    // some code
+  }
+}
+
+function doRequest(url, timeout = 2000, callback = someCallback()) {
+  // rest of code
+}
+```
+
+
+#### Block Level Functions
+
+
+In ES5 strict mode, a function declaration made within a block would throw an
+error:
+
+```javascript
+"use strict";
+
+if (true) {
+  // throws a syntax error
+  function someThing() {
+    // some code
+  }
+}
+```
+
+In ES6, you can declare a function within a block and it will only be accessible
+within that block:
+
+```javascript
+"use strict";
+
+if (true) {
+  console.log(typeof someThing()); // "function"
```

Comment:
I believe you should remove the inner parentheses here (+3 similar lines below).